matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/SyncResponsePostTreatmentAggregatorHandler.kt

```diff
@@ -92,6 +94,6 @@ internal class SyncResponsePostTreatmentAggregatorHandler @Inject constructor(
 
     private suspend fun fetchUsers(usersToFetch: MutableList<String>) = usersToFetch.map {
         val profileJson = profileService.getProfile(it)
-        MatrixItem.UserItem.fromJson(it, profileJson)
+        User.fromJson(it, profileJson).toMatrixItem()
```

Comment:
As I said, we should not use `MatrixItem` internally.
Can you change to:
```kotlin
User.fromJson(it, profileJson)
```
update the code to make it compile and remove `UserEntityFactory.create(userItem: MatrixItem.UserItem): UserEntity`

Thanks!

Reply:
Ahh I see what you mean! Sorry I misunderstood the first time